File: server_asp/src/models/ClientAccount.cs
using System;
using System.Collections.Generic;

namespace shieldtify.models;

public partial class ClientAccount
{
    public Guid Uid { get; set; }

    public string Username { get; set; } = null!;

    public string Password { get; set; } = null!;

    public string DisplayName { get; set; } = null!;

    public string Email { get; set; } = null!;

    public DateTime? ChangedPasswordAt { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }


    public virtual ICollection<CartItem> CartItems { get; set; } = new List<CartItem>();

    public virtual ICollection<ClientAddress> ClientAddresses { get; set; } = new List<ClientAddress>();

    public virtual ICollection<Conversation> Conversations { get; set; } = new List<Conversation>();

    public virtual ICollection<Order> Orders { get; set; } = new List<Order>();

    public virtual ICollection<Post> Posts { get; set; } = new List<Post>();

    public virtual ICollection<Vote> Votes { get; set; } = new List<Vote>();
}

File: server_nodejs/src/models/client_account.js
// Client account (UID, username, password, display_name, email)
module.exports = (sequelize, Sequelize) => {
    class ClientAccount extends Sequelize.Model {}

    ClientAccount.init(
        {
            uid: {
                type: Sequelize.UUID,
                primaryKey: true,
            },
            username: {
                type: Sequelize.STRING,
                allowNull: false,
            },
            password: {
                type: Sequelize.STRING,
                allowNull: false,
            },
            display_name: {
                type: Sequelize.STRING,
                allowNull: false,
            },
            email: {
                type: Sequelize.STRING,
                allowNull: false,
            },
            changed_password_at: {
                type: Sequelize.DATE,
                allowNull: true,
            },
        },
        {
            sequelize,
            modelName: 'client_account',
            timestamps: true,
            paranoid: false,
            underscored: true,
        },
    );

    ClientAccount.associate = (models) => {
        ClientAccount.hasMany(models.post, {
            foreignKey: 'created_by',
            as: 'posts',
        });
        ClientAccount.hasMany(models.cart_item, {
            foreignKey: 'clientid',
            as: 'cart_item',
        });
        ClientAccount.hasMany(models.client_address, {
            foreignKey: 'clientid',
            as: 'client_address',
        });
        ClientAccount.hasMany(models.conversation, {
            foreignKey: 'clientid',
            as: 'conversation',
        });
        ClientAccount.hasMany(models.order, {
            foreignKey: 'clientid',
            as: 'order',
        });
        ClientAccount.hasMany(models.vote, {
            foreignKey: 'clientid',
            as: 'vote',
        });
    };
    return ClientAccount;
};
